hack/check-conflicts.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -e
+
+echo -e "\nChecking for presence of conflict notes in source files..."
+
+overall_failed=0
+
+check_results(){
+    if [ -z "$1" ]; then
+        echo PASS
+    else
+        echo FAIL
+        echo ""
+        echo "$1"
+        echo ""
+        return 1
+    fi
+    return 0
+}
+
+echo -ne "\n\tChecking staged files... "
+if [ $(git diff --cached --name-only | wc -l) != "0" ]; then
+    results=$(git diff --cached --name-only | grep -v "vendor/" | xargs grep -PiIrnH '<<<<''<<<|>>>''>>>>|===''====' || true)
```

Comment:
we can move even check logic into `check_result` (or whatever we call it later on) function. Something along these lines:

```
check_merge_conflicts() {
  for f in "$@"; do 
    grep -PiIrnH '<<<<''<<<|>>>''>>>>|===''====' "$f" && echo "$f" && break
  done || found=1
if [ -z "$found" ]; then
  echo PASS
else
  echo FAIL
  exit 1
fi
}

check_merge_conflicts $(git diff --cached --name-only | grep -v "vendor/")
check_merge_conflicts $(git ls-tree --full-tree -r HEAD --name-only | grep -v "vendor/")
```